File: src/main/java/com/learn/todolist/api/controller/UserRegisterController.java
package com.learn.todolist.api.controller;

import com.learn.todolist.api.dto.UserLoginReq;
import com.learn.todolist.api.dto.UserRegisterReq;
import com.learn.todolist.application.CommonResponse;
import com.learn.todolist.application.UserApplication;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/v1/registers")
public class UserRegisterController {
    private final UserApplication userApplication;

    @PostMapping("/register")
    public ResponseEntity<CommonResponse> register(@RequestBody UserRegisterReq request) {
        CommonResponse response = userApplication.processRegister(request);
        return ResponseEntity.ok(response);
    }

    @DeleteMapping("/withdraw")
    public ResponseEntity<CommonResponse> withdrawal(@RequestBody UserLoginReq request) {
        CommonResponse response = userApplication.processWithdrawal(request);
        return ResponseEntity.ok(response);
    }
}

File: src/main/kotlin/com/todolist/aplication/user/controller/UserRegisterController.kt
package com.todolist.aplication.user.controller

import com.todolist.aplication.user.dto.request.UserLoginRequest
import com.todolist.aplication.user.dto.request.UserRegisterRequest
import com.todolist.aplication.user.usecase.UserApplication
import com.todolist.domain.common.CommonResponse
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1")
class UserRegisterController(
    private val userApplication: UserApplication
) {
    @PostMapping("/registers/register")
    fun register(@RequestBody request: UserRegisterRequest): ResponseEntity<CommonResponse> {
        val response = userApplication.isRegister(request)
        return ResponseEntity(response, HttpStatus.OK)
    }

    @DeleteMapping("/registers/withdrawal")
    fun withdrawal(@RequestBody request: UserLoginRequest): ResponseEntity<CommonResponse> {
        val response = userApplication.isWithdrawal(request)
        return ResponseEntity(response, HttpStatus.OK)
    }
}